### ISSUE TITLE
DatePickerDialog to display the selected date when reopened

### ISSUE THREAD

**Issue by spykeburn** [wdullaer/MaterialDateTimePicker#127]
A date once selected  doesn't show highlighted when the dialog is opened again. The cursor goes back to the current date. Is there a way we can do this? I am guessing it has to do something with the onResume method.

                     dateButton.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                Calendar now = Calendar.getInstance();
                DatePickerDialog dpd = DatePickerDialog.newInstance(
                        MainActivity.this,
                        now.get(Calendar.YEAR),
                        now.get(Calendar.MONTH),
                        now.get(Calendar.DAY_OF_MONTH)
                );
             
                dpd.show(getFragmentManager(), "Datepickerdialog");
    
    }
                public void onResume() {
                super.onResume();

        DatePickerDialog dpd = (DatePickerDialog) getFragmentManager().findFragmentByTag("Datepickerdialog");

        if(dpd != null) dpd.setOnDateSetListener(this);
    }

                   public void onDateSet(DatePickerDialog view, int year, int monthOfYear, int dayOfMonth) {
        String date = "You picked the following date: "+dayOfMonth+"/"+(++monthOfYear)+"/"+year;
        dateTextView.setText(date);
    }

**Comment by wdullaer** [comment]
I don't fully understand what you mean with "reopened".

When you just do an orientation change, and you set the picker up as in the example app, it will remember the selected date.

If you mean someone selecting a date, hitting ok and then reopening the dialog, your app will be responsible to save the user selection and set it as the new default selection when reopening it.

**Comment by hitchan** [comment]
Why don't you just store a Calendar instance with the selected date and set it to that when you go to open it again?

**Comment by wdullaer** [comment]
I still do not understand your usecase.

If the dialog gets recreated due to an orientation/configuration change, it will properly restore it's current state.

If the user dismisses/cancels/oks the dialog it'll call the respective callback and then ceases to exist.
There is nothing which save or restore the state.
It's the job of your application to make sure the next dialog gets initialised to the correct date.

In short: I don't properly understand your usecase and hence can't really see where the limitation / issue lies.

**Comment by wdullaer** [comment]
You should not try to reuse a previous instance of the Dialog after it has been dismissed.
It is nearly impossible for the library to keep its state when it's not shown without additional action by the application developer, at which point you could just well create a new instance each time you need to show the dialog.

**Comment by mmounirf** [comment]
Simply, don't use this line
`Calendar now = Calendar.getInstance();`
every time you want to set time or date. You can define **now** as a public variable and reuse it whenever you want to get date/time from calendar.
